source/applications/dependency-injection.md

```diff
@@ -209,3 +215,28 @@ export default {
   initialize: initialize
 };
 ```
+
+### Getting an Application Instance from a Factory Instance
+
+[`Ember.getOwner`][4] will retrieve the application instance from an Ember
+Object that was instantiated from a registered factory. This means that
+framework objects like components, helpers, and routes can use
+[`Ember.getOwner`][4] to perform lookups through their application instance.
+
+```app/helpers/logger.js
+import Ember from 'ember';
+
+const { getOwner } = Ember;
+
+export default Ember.Helper.extend({
+  init() {
+    this._super(...arguments);
+
+    this.errorReporter = getOwner(this).lookup('error-reporter:sentry');
```

Comment:
A good example here might be to get the name of the error reporting service from an argument to the component:

```js
// Usage: {{logger 'error!!' reportWith='sentry'}}
import Ember from 'ember';
const { getOwner, Helper } = Ember;

export default Helper.extend({
  compute([message], {reportWith}) {
    let errorReportingServiceName = `service:${reportWith}`;
    let applicationInstance = getOwner(this);
    let errorReporter = applicationInstance.lookup(errorReportingServiceName);
    errorReporter.report(message);
  }
});
```

Even better might be an example that did currency formatting or date formatting. Generally you do not want helpers to do things with side-effects (like log), and thus the example of a logger might be an anti-pattern.